src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +24,25 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> text = new TreeSet<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                for (Character x : line.toLowerCase().replaceAll("[^a-zA-Z]", "").toCharArray()) {
+                    text.add(x);
+                }
+            }
+        } catch (IOException e) {
+            throw new FileNotFoundException();
```

Comment:
it's not always will be FileNotFoudnException, but you won't know it. We can just leave throws near method signature in this case